tox.ini

```diff
@@ -1,7 +1,7 @@
 [tox]
 env_list =  # Environments to run when called with no parameters.
     lint-{black,ruff,pyright,shellcheck,codespell,docs}
-    test-{py38,py310,py311,py312}
```

Comment:
how come?

Reply:
Craft-providers doesn't support Python 3.12: https://github.com/canonical/craft-providers/issues/317

I've made an issue for re-adding it: https://github.com/canonical/craft-application/issues/32